sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp

```diff
@@ -1,5 +1,9 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
+#if !defined(INCLUDE_AZURE_CORE_EXTENDABLE_ENUMERATION_HPP)
+#pragma once
+
+#define INCLUDE_AZURE_CORE_EXTENDABLE_ENUMERATION_HPP
 
 /**
  * @file
```

Comment:
BTW, in the code below, code snippet is incorrect (incomplete): statics must be marked with `..._DLLEXPORT`, or they won't link when Azure SDK is built as DLL.

Reply:
If I added in the DLLEXPORT goo, it would unnecessarily complicate the example. Examples should demonstrate the pattern without any complicated goo.

And the example works just fine if it's not used in a header.